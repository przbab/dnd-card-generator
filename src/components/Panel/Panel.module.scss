.root {
    box-shadow: 1px 3px 5px #ddd;
}
