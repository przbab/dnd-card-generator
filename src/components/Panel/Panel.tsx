import React from 'react';
import classnames from 'classnames';
import styles from './Panel.module.scss';

export namespace Panel {
    export type Props = {
        children: React.ReactElement;
        className?: string;
    };
}

export function Panel({ children, className }: Panel.Props) {
    return <div className={classnames(styles.root, className)}>{children}</div>;
}

Panel.defaultProps = {className: '',};
