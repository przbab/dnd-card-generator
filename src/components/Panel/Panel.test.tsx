import React from 'react';
import renderer from 'react-test-renderer';
import { Panel } from './Panel';

describe('components', () => {
    describe('Panel', () => {
        test('Should match snapshot', () => {
            const tree = renderer
                .create(
                    <Panel className="additional-class-name">
                        <span>Test children</span>
                    </Panel>
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
