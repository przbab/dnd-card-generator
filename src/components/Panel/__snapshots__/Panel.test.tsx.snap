// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Panel Should match snapshot 1`] = `
<div
  className="root additional-class-name"
>
  <span>
    Test children
  </span>
</div>
`;
